src/main/java/org/jenkinsci/plugins/workflow/cps/DSL.java

```diff
@@ -104,12 +106,15 @@ protected Object readResolve() throws IOException {
         return this;
     }
 
-    /** Allows user to disable saving step info, if this imposes too high a performance overhead
-     *   by setting System Property org.jenkinsci.plugins.workflow.cps.DSL.keepStepInfo to 'false'.
-     *  This may be modified at runtime by the user if needed for debugging. */
-    private static boolean isKeepStepInfo() {
-        String prop = System.getProperty(DSL.class.getName()+".keepStepInfo");
-        return (StringUtils.isEmpty(prop)|| Boolean.parseBoolean(prop));
+    private static final String KEEP_STEP_ARGUMENTS_PROPERTYNAME = (DSL.class.getName()+".keepStepArguments");
+
+    private static boolean isKeepStepArguments = StringUtils.isEmpty(System.getProperty(KEEP_STEP_ARGUMENTS_PROPERTYNAME))
+            || Boolean.parseBoolean(System.getProperty(KEEP_STEP_ARGUMENTS_PROPERTYNAME));
+
+    /** Tell us if we should store {@link Step} arguments in an {@link org.jenkinsci.plugins.workflow.actions.ArgumentsAction}
+     *  or simply discard them (if set to false, explicitly) */
+    public static boolean isKeepStepArguments() {
```

Comment:
@jglick While groovy can be used to mutate this since it ignores access permissions, we don't want to expose the field to normal mutation from java -- but we do want to be able to read it potentially.